### DISCUSSION TITLE
ReferenceError: File is not defined

### DISCUSSION THREAD

**Discussion by raphaelmans** [vercel/next.js#56032] [Help]
### Summary

Fail on build

```
Error occurred prerendering page "/profile/hustler". Read more: https://nextjs.org/docs/messages/prerender-error
ReferenceError: File is not defined
```

```
> Export encountered errors on following paths:
        /(logged)/onboarding/business/page: /onboarding/business
        /(logged)/profile/business/page: /profile/business
        /(logged)/profile/hustler/page: /profile/hustler
```

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103
Binaries:
  Node: 18.13.0
  npm: 8.19.3
  Yarn: 1.22.19
  pnpm: 8.6.12
Relevant Packages:
  next: 13.5.3
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: N/A
Next.js Config:
  output: N/A
```

**Comment by ealexhaywood** [comment]
I was getting this error on a page that was using `FormData` and `File`, everything worked locally on `next dev` and `next build && next start`, but was failing when deploying to Vercel.

Upgrading to Node 20 fixed it

**Reply by timagixe** [reply]
Worked for me too, thanks!

**Reply by timagixe** [reply]
@raphaelmans could you please mark this as a solution?

**Reply by smukideejeah** [reply]
It's working. Thanks.

**Reply by vishnukumar89** [reply]
@ealexhaywood I have Node : v20.13.1. Still, I'm facing the same issue.

**Reply by netrolite** [reply]
thank you so much

**Reply by nduongg04** [reply]
thanks for your anwser

**Reply by btnalexandre** [reply]
thanks work like a charm !

**Reply by dragonfore4** [reply]
This work for me too , thanks

**Reply by ChanceVoightDev** [reply]
How absolutely ridiculous. Thank you! And not thank you Vercel.

**Reply by thesidshah** [reply]
You sir are a god send

**Comment by devslife7** [comment]
I got the same error when deploying. It works great on my machine because I am running node 20x. Vercel deployment settings out of the box uses node 18x. So i went to vercel app settings and changed it to 20x, not it works like a charm.

**Reply by steinhardt21** [reply]
Also for me! Thank you!

**Reply by rashgaroth** [reply]
> I got the same error when deploying. It works great on my machine because I am running node 20x. Vercel deployment settings out of the box uses node 18x. So i went to vercel app settings and changed it to 20x, not it works like a charm.

got the same issue on Amplify, and worked when i've changed the node version to 20x.

**Comment by ahmadalfy** [comment]
Can confirm, migrating to Node 20 fixed the issue

**Reply by XoL1507** [reply]
tell ma. maybe i can help.

**Reply by smukideejeah** [reply]
I confirm.

**Comment by faustinozanetto** [comment]
I've encountered the same issue and I can confirm upgrading to Node 20 fixes it.

**Comment by softmarshmallow** [comment]
that was easy

**Comment by arjendevos** [comment]
I upped to `v20` but still got the issue: `File is not defined`

**Comment by bascor1207** [comment]
The same issue occurred, and the same resolution applied. It was failing in my GitHub Actions, but upgrading to Node 20.x in main.yml solved the issue.

**Comment by henryobiaraije** [comment]
Upgrading to v20 solved this, even locally.
